alacritty_terminal/src/term/mod.rs

```diff
@@ -1813,103 +1813,216 @@ impl<T: EventListener> Handler for Term<T> {
+                    },
+                    NamedPrivateMode::ReportFocusInOut => {
+                        self.mode.contains(TermMode::FOCUS_IN_OUT)
+                    },
+                    NamedPrivateMode::Utf8Mouse => self.mode.contains(TermMode::UTF8_MOUSE),
+                    NamedPrivateMode::SgrMouse => self.mode.contains(TermMode::SGR_MOUSE),
+                    NamedPrivateMode::AlternateScroll => {
+                        self.mode.contains(TermMode::ALTERNATE_SCROLL)
+                    },
+                    NamedPrivateMode::UrgencyHints => self.mode.contains(TermMode::URGENCY_HINTS),
+                    NamedPrivateMode::SwapScreenAndSetRestoreCursor => {
+                        self.mode.contains(TermMode::ALT_SCREEN)
+                    },
+                    NamedPrivateMode::BracketedPaste => {
+                        self.mode.contains(TermMode::BRACKETED_PASTE)
+                    },
+                    NamedPrivateMode::SyncUpdate => false,
+                    NamedPrivateMode::ColumnMode => unreachable!(),
+                };
+
+                if set {
+                    1
+                } else {
+                    2
+                }
+            },
+        };
+
+        self.event_proxy.send_event(Event::PtyWrite(format!("\x1b[?{};{status}$p", mode.raw())));
+    }
+
+    #[inline]
+    fn set_mode(&mut self, mode: ansi::Mode) {
+        let ansi::Mode::Named(mode) = mode else {
+            debug!("Unsetting unknown public mode {}", mode.raw());
+            return;
+        };
+
+        trace!("Setting public mode: {:?}", mode);
+        match mode {
+            NamedMode::Insert => self.mode.insert(TermMode::INSERT),
+            NamedMode::LineFeedNewLine => self.mode.insert(TermMode::LINE_FEED_NEW_LINE),
+        }
+    }
+
+    #[inline]
+    fn unset_mode(&mut self, mode: ansi::Mode) {
+        let ansi::Mode::Named(mode) = mode else {
+            debug!("Unsetting unknown public mode {}", mode.raw());
+            return;
+        };
+
+        trace!("Setting public mode: {:?}", mode);
+        match mode {
+            NamedMode::Insert => {
+                self.mode.remove(TermMode::INSERT);
+                self.mark_fully_damaged();
+            },
+            NamedMode::LineFeedNewLine => self.mode.remove(TermMode::LINE_FEED_NEW_LINE),
         }
     }
 
+    #[inline]
+    fn report_mode(&mut self, mode: ansi::Mode) {
+        trace!("Reporting mode {mode:?}");
+        let status = match mode {
+            ansi::Mode::Named(mode) => {
+                let set = match mode {
+                    NamedMode::Insert => self.mode.contains(TermMode::INSERT),
+                    NamedMode::LineFeedNewLine => self.mode.contains(TermMode::LINE_FEED_NEW_LINE),
+                };
+
+                if set {
+                    1
+                } else {
+                    2
+                }
+            },
+            ansi::Mode::Unknown(_) => 0,
```

Comment:
I'd like to have this consistently above/below the named modes with private/public modes. Where it is I don't care, but having it at the top for one and the bottom for the other is just confusing.